src/org/exist/storage/NativeBroker.java

```diff
@@ -2847,6 +2851,39 @@ private void dropIndex(final Txn transaction, final DocumentImpl document) throw
         getIndexController().flush();
     }
 
+    private void dropDomNodes(final Txn transaction, final DocumentImpl document) {
+        try {
+            if(!document.getMetadata().isReferenced()) {
+                new DOMTransaction(this, domDb, LockMode.WRITE_LOCK) {
+                    @Override
+                    public Object start() {
+                        final NodeHandle node = (NodeHandle) document.getFirstChild();
+                        domDb.removeAll(transaction, node.getInternalAddress());
+                        return null;
+                    }
+                }.run();
+            }
+        } catch(NullPointerException npe0) {
+            LOG.error("Caught NPE in DOMTransaction to actually be able to remove the document.");
```

Comment:
as there is no recovery needed, better to log as WARNing ?

Reply:
This code was just moved. TBH I have no idea why we try and catch an NPE there, I assumed there was some reason that was forgotten in the past.

Reply:
@shabanovd Ah okay. Thanks :-)

Reply:
@dizzzz seems like a legit error to me.